libyul/AsmParser.cpp

```diff
@@ -263,36 +250,32 @@ Expression Parser::parseExpression()
 {
 	RecursionGuard recursionGuard(*this);
 
-	ElementaryOperation operation = parseElementaryOperation();
-	if (holds_alternative<FunctionCall>(operation) || currentToken() == Token::LParen)
-		return parseCall(std::move(operation));
-	else if (holds_alternative<Identifier>(operation))
-		return std::get<Identifier>(operation);
+	ElementaryOperation operation = parseLiteralOrIdentifier();
+	if (holds_alternative<Identifier>(operation))
+	{
+		if (currentToken() == Token::LParen)
+			return parseCall(std::move(operation));
+		auto identifier = std::get<Identifier>(operation);
+		if (m_dialect.builtin(identifier.name))
+			fatalParserError(7104_error, "Builtin function \"" + identifier.name.str() + "\" must be called.");
```

Comment:
Not sure we are objectively better with this here, but it is kind of a logical counterpart to the one in the assignment case in `parseStatement`.